pkg/apis/k8sbundle/v1alpha1/bundle_types.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// UnstructuredJSON represents JSON not bound to any particular schema
+type UnstructuredJSON map[string]interface{}
```

Comment:
We have RawExtension for this. You can't use interface{} in an api type.

Reply:
Pardon the ignorance, but is there a technical reason why you can't use interface{}?

Reply:
Well, the tooling doesn't support it, for one :)

But I think that's more or less a deliberate choice?
* there's no way to document for users what they're actually communicating to the server
* there's no way to promise things that process API types that they're guaranteed to only get safe things inside the interface{}'s (type safety)

Reply:
Isn't the same also true for RawExtension?

Reply:
In any case, I think I would add this in a separate PR and ensure that proto and this file are in-sync. I'm warming to the RawExtension type and I like the idea of compatibility with the broader K8S stuff.